File: jenkins-plugin/src/main/java/org/eazyportal/plugin/release/jenkins/action/SetReleaseVersionActionFactory.java
package org.eazyportal.plugin.release.jenkins.action;

import com.google.inject.Inject;
import hudson.Extension;
import hudson.model.InvisibleAction;
import org.eazyportal.plugin.release.core.action.SetReleaseVersionAction;
import org.eazyportal.plugin.release.core.version.ReleaseVersionProvider;
import org.eazyportal.plugin.release.core.version.VersionIncrementProvider;
import org.eazyportal.plugin.release.jenkins.ReleaseStepConfig;

import java.io.Serializable;

@Extension
public class SetReleaseVersionActionFactory extends InvisibleAction implements Serializable {

    @Inject
    private transient ReleaseStepConfig releaseStepConfig;
    @Inject
    private transient ReleaseVersionProvider releaseVersionProvider;
    @Inject
    private transient VersionIncrementProvider versionIncrementProvider;

    public SetReleaseVersionAction create() {
        return new SetReleaseVersionAction(
            releaseStepConfig.getConventionalCommitTypes(),
            releaseVersionProvider,
            releaseStepConfig.getScmActions(),
            releaseStepConfig.getScmConfig(),
            versionIncrementProvider
        );
    }

}

File: gradle-plugin/src/main/kotlin/org/eazyportal/plugin/release/gradle/action/SetReleaseVersionActionFactory.kt
package org.eazyportal.plugin.release.gradle.action

import org.eazyportal.plugin.release.core.action.SetReleaseVersionAction
import org.eazyportal.plugin.release.core.version.ReleaseVersionProvider
import org.eazyportal.plugin.release.core.version.VersionIncrementProvider
import org.eazyportal.plugin.release.gradle.model.EazyReleasePluginExtension

class SetReleaseVersionActionFactory : ReleaseActionFactory<SetReleaseVersionAction> {

    override fun create(extension: EazyReleasePluginExtension): SetReleaseVersionAction =
        SetReleaseVersionAction(
            extension.conventionalCommitTypes,
            extension.projectActionsFactory,
            ReleaseVersionProvider(),
            extension.scmActions,
            extension.scmConfig,
            VersionIncrementProvider()
        )

}